apps/www/public/r/styles/default/use-composition.json

```
@@ -0,0 +1,12 @@
+{
+  "name": "use-composition",
+  "type": "registry:hook",
+  "files": [
+    {
+      "path": "hooks/use-composition.tsx",
+      "content": "import * as React from \"react\"\n\ntype ReactElementWithDisplayName = React.ReactElement & {\n  type: { displayName: string }\n}\n\nconst isElement = (\n  element: unknown | undefined\n): element is ReactElementWithDisplayName => {\n  return (\n    (element as ReactElementWithDisplayName)?.type?.displayName !== undefined\n  )\n}\n\nexport function useComposition(\n  children: React.ReactNode,\n  component: string\n): Array<React.ReactNode> {\n  const Children = React.useMemo(\n    () =>\n      React.Children.toArray(children).find((child) => {\n        if (isElement(child)) return child.type.displayName === component\n        return false\n      }),\n    [children, component]\n  ) as Array<React.ReactNode>\n\n  return Children ?? []\n}\n",
```

Shouldn't this use `filter` instead of `find`?
```suggestion
      "content": "import * as React from \"react\"\n\ntype ReactElementWithDisplayName = React.ReactElement & {\n  type: { displayName: string }\n}\n\nconst isElement = (\n  element: unknown | undefined\n): element is ReactElementWithDisplayName => {\n  return (\n    (element as ReactElementWithDisplayName)?.type?.displayName !== undefined\n  )\n}\n\nexport function useComposition(\n  children: React.ReactNode,\n  component: string\n): Array<React.ReactNode> {\n  const Children = React.useMemo(\n    () =>\n      React.Children.toArray(children).filter((child) => {\n        if (isElement(child)) return child.type.displayName === component\n        return false\n      }),\n    [children, component]\n  ) as Array<React.ReactNode>\n\n  return Children ?? []\n}\n",
```

---

idea is to only allow 1 icon per side, doesn't make sense to me (yet?) when you need multiple icons on the same side. 

left executes a find & right executes a find :)

---

Okay, didn't dig into it too much. Was attempting to use this code, and was running into an error, since the casting as array didn't appear to be working. Getting this error:
```
TypeError: s.some is not a function. (In 's.some(p=>p.props.side==="left")', 's.some' is undefined)
```

The value of `Children` ends up being:
```
{$$typeof: Symbol(react.transitional.element), type: Object, key: ".0", ref: null, props: Object}
```